lib/internal/webstreams/readablestream.js

```diff
@@ -1424,13 +1424,15 @@ function readableStreamTee(stream, cloneForBranch2) {
         });
       },
       [kClose]() {
-        reading = false;
-        if (!canceled1)
-          readableStreamDefaultControllerClose(branch1[kState].controller);
-        if (!canceled2)
-          readableStreamDefaultControllerClose(branch2[kState].controller);
-        if (!canceled1 || !canceled2)
-          cancelPromise.resolve();
+        process.nextTick(() => {
```

Comment:
No, I meant that opposite. I think https://github.com/nodejs/node/blob/master/lib/internal/streams/end-of-stream.js should be `queueMicrotask` instead of `process.nextTick` but I'm not sure 🤔

Reply:
I did a simple test by just calling `console.log` on `eos` and it doesn't call, so not sure either if the end-of-stream is the best place. Looks like the `kClose` is suitable for that otherwise my second comment https://github.com/nodejs/node/pull/40901#issuecomment-975741033 is valid.

Reply:
Please add a comment here explaining that the addition of the process.nextTick is not part of the spec and why it's needed with a link back to the original issue.

Reply:
Done.